react/components/Iframe.js

```diff
@@ -119,32 +119,33 @@ class Iframe extends Component {
         : search.includes('env=beta')
           ? search
           : search + '&env=beta'
-    const nextSrc = this.buildSrc(nextProps, patchedSearch, hash)
+    return `/admin/${props.isDeloreanAdmin ? 'iframe' : 'app'}/${props.params.slug || ''}${patchedSearch || ''}${hash || ''}`
+  }
 
-    return nextProps.isDeloreanAdmin || !(this.iframe && this.iframe.contentWindow.location.pathname.startsWith(nextSrc))
+  componentDidUpdate() {
+    const nextSrc = this.buildSrc(this.props)
+    if (this.props.isDeloreanAdmin || !(this.iframe && this.iframe.contentWindow.location.pathname === nextSrc)){
+      this.iframe && this.iframe.contentWindow.location.replace(nextSrc)
+    }
   }
 
+  fixedSrc = ''
+
   render() {
     const { isDeloreanAdmin, params: { slug = '' } } = this.props
     if (startsWith(isDeloreanAdmin ? 'iframe/' : 'app/', slug)) {
       return
     }
     const { loaded } = this.state
-    const { location } = window
-
-    const hash = loaded ? location.hash : ''
-    const search = loaded ? location.search || '' : ''
-    const env = getEnv()
-    const patchedSearch = env !== 'beta'
-      ? search
-      : search === ''
-        ? '?env=beta'
-        : search.includes('env=beta')
-          ? search
-          : search + '&env=beta'
-
-    const src = location.pathname.replace('/admin', isDeloreanAdmin ? '/admin/iframe' : '/admin/app') + patchedSearch + hash
 
+    // if we update the iframe src, the iframe redirects to the src address and it automatically propagates this navigation to
+    // the browser history (this is the expected behaviour https://trillworks.com/nick/2014/06/11/changing-the-src-attribute-of-an-iframe-modifies-the-history/)
+    // so we only want to set the iframe src when it has been created in order to start on the URL that has been inserted in the browser in the first navigation.
+    // the navigation from outside into the iframe is made by `componentDidUpdate` changing the window location, this doesnt affect the browser history.
+    if (this.fixedSrc === '') {
+      this.fixedSrc = this.buildSrc(this.props)
```

Comment:
Changing to different admins has no consequence? Nice

Reply:
I would probably prefer something like returning false on `shouldComponentUpdate` if the iframe has been mounted, but this is not a major issue. Go ahead, merge this and let's test it out.